components/engine_rocks/src/event_listener.rs

```diff
@@ -42,6 +56,19 @@ impl rocksdb::EventListener for RocksEventListener {
                 &info.compaction_reason().to_string(),
             ])
             .inc();
+        if get_io_type() == IOType::Compaction {
+            set_io_type(IOType::Other);
+        }
+    }
+
+    fn on_subcompaction_begin(&self, _info: &SubcompactionJobInfo) {
+        set_io_type(IOType::Compaction);
+    }
+
+    fn on_subcompaction_completed(&self, _info: &SubcompactionJobInfo) {
+        if get_io_type() == IOType::Compaction {
+            set_io_type(IOType::Other);
```

Comment:
`Other` is tagged for some missing tagged threads or when an error occurs for the io. I'm not sure whether it's better to tag `Other` here.

Reply:
The IO type after compaction is unspecified. For instance flush can be scheduled to low-priority threads used for compaction before.

Reply:
I mean just keep the type, future flush or compaction will set the type at the beginning anyway.

Reply:
True. But I still don't see a strong enough reason to make the compromise. It's just *more correct* to set IO type to unknown than leaving it potentially wrong.

Reply:
Alright, let's keep it